docs/topics/whatsnew21.md

```diff
@@ -10,10 +23,1562 @@ All you need to do is [change the Kotlin version](configure-build-for-eap.md) to
+
+Here's an example demonstrating the change:
+
+```kotlin
+sealed class Result
+object Error: Result()
+class Success(val value: String): Result()
+
+fun <T : Result> render(result: T) = when (result) {
+    Error -> "Error!"
+    is Success -> result.value
+    // Requires no else branch
+}
+```
+
+## Kotlin K2 compiler
+
+With Kotlin 2.1.0, the K2 compiler now provides [more flexibility when working with compiler checks](#extra-compiler-checks)
+and [warnings](#global-warning-suppression), as well as [improved support for the kapt plugin](#improved-k2-kapt-implementation).
+
+### Extra compiler checks
+
+With Kotlin 2.1.0, you can now enable additional checks in the K2 compiler.
+These are extra declaration, expression, and type checks that are usually not crucial for compilation
+but can still be useful if you want to validate the following cases:
+
+| Check type                                            | Comment                                                                                  |
+|-------------------------------------------------------|------------------------------------------------------------------------------------------|
+| `REDUNDANT_NULLABLE`                                  | `Boolean??` is used instead of `Boolean?`                                                |
+| `PLATFORM_CLASS_MAPPED_TO_KOTLIN`                     | `java.lang.String` is used instead of `kotlin.String`                                    |
+| `ARRAY_EQUALITY_OPERATOR_CAN_BE_REPLACED_WITH_EQUALS` | `arrayOf("") == arrayOf("")` is used instead of `arrayOf("").contentEquals(arrayOf(""))` |
+| `REDUNDANT_CALL_OF_CONVERSION_METHOD`                 | `42.toInt()` is used instead of `42`                                                     |
+| `USELESS_CALL_ON_NOT_NULL`                            | `"".orEmpty()` is used instead of `""`                                                   |
+| `REDUNDANT_SINGLE_EXPRESSION_STRING_TEMPLATE`         | `"$string"` is used instead of `string`                                                  |
+| `UNUSED_ANONYMOUS_PARAMETER`                          | A parameter is passed in the lambda expression but never used                            |
+| `REDUNDANT_VISIBILITY_MODIFIER`                       | `public class Klass` is used instead of `class Klass`                                    |
+| `REDUNDANT_MODALITY_MODIFIER`                         | `final class Klass` is used instead of `class Klass`                                     |
+| `REDUNDANT_SETTER_PARAMETER_TYPE`                     | `set(value: Int)` is used instead of `set(value)`                                        |
+| `CAN_BE_VAL`                                          | `var local = 0` is defined but never reassigned, can be `val local = 42` instead         |
+| `ASSIGNED_VALUE_IS_NEVER_READ`                        | `val local = 42` is defined but never used afterward in the code                         |
```

Comment:
```suggestion
| `ASSIGNED_VALUE_IS_NEVER_READ`                        | `val local = 42` is defined but never used in the code                         |
```

Reply:
For conciseness and consitency with the next one

Reply:
No, there's a difference. Otherwise, they are identical and there would be no need for two different warnings

Reply:
Not sure about this one, otherwise, the idea of the check is not clear.